bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/NextGenDMImageURIBuilder.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    /**
+     *  Image height
+     * @param height - an integer.
+     */
+    public NextGenDMImageURIBuilder withHeight(int height) {
+        this.height = height;
+        return this;
+    }
+
+
+    /**
+     * Set to use webp image format.
+     * @param preferWebp - should set preferwebp param.
+     */
+    public NextGenDMImageURIBuilder withPreferWebp(boolean preferWebp) {
+        this.preferWebp = preferWebp;
+        return this;
+    }
+
+    /**
+     * Use this to create a NextGen Dynamic Media Image URI.
+     * @return a uri.
+     */
+    public String build() {
+        if(StringUtils.isNotEmpty(this.fileReference) && this.config != null) {
+            Scanner scanner = new Scanner(this.fileReference);
+            scanner.useDelimiter("/");
+            String assetId = scanner.next();
+            scanner = new Scanner(scanner.next());
+            scanner.useDelimiter("\\.");
+            String assetName = scanner.hasNext() ? scanner.next() : assetId;
+            String assetExtension = scanner.hasNext() ? scanner.next() : DEFAULT_NGDM_ASSET_EXTENSION;
+            String imageDeliveryBasePath = this.config.getImageDeliveryBasePath();
+            String imageDeliveryPath = imageDeliveryBasePath.replace(PATH_PLACEHOLDER_ASSET_ID, assetId);
+            imageDeliveryPath = imageDeliveryPath.replace(PATH_PLACEHOLDER_SEO_NAME, assetName);
+            imageDeliveryPath = imageDeliveryPath.replace(PATH_PLACEHOLDER_FORMAT, assetExtension);
+            String repositoryId = this.config.getRepositoryId();
+            StringBuilder uriBuilder = new StringBuilder("https://" + repositoryId + imageDeliveryPath);
+            Map<String, String> params = new HashMap<>();
+            if(this.width > 0) {
+                params.put("width", Integer.toString(this.width));
+            }
+            if(this.height > 0) {
+                params.put("height", Integer.toString(this.height));
+            }
+            if(this.preferWebp) {
+                params.put("preferwebp", "true");
+            }
+            if (StringUtils.isNotEmpty(this.smartCropAspectRatio)) {
+                if (isValidSmartCrop(this.smartCropAspectRatio)) {
+                    params.put("crop", String.format("%s,smart", this.smartCropAspectRatio));
+                } else {
+                    LOGGER.error("Invalid smartCrop value at {}", this.smartCropAspectRatio);
+                }
+            }
+            if(params.size() > 0) {
+                uriBuilder.append("?");
+                for(Map.Entry<String, String> entry: params.entrySet()) {
+                    uriBuilder.append(entry.getKey() + "=" + entry.getValue());
+                    uriBuilder.append("&");
+                }
+                uriBuilder.deleteCharAt(uriBuilder.length() - 1);
+            }
+            return uriBuilder.toString();
+        }
+        LOGGER.error("Invalid fileReference or NGDMConfig. fileReference = {}", this.fileReference);
```

Comment:
Don't log an error for this because it may generate bogus alerts on AEM CS . You may log the problem on debug level or ignore it.

Reply:
changed to log.info